src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests.cs

```diff
@@ -200,5 +201,96 @@ class C
             }
             EOF();
         }
+
+        [Fact, WorkItem(14785, "https://github.com/dotnet/roslyn/issues/14785")]
+        public void IsPatternPrecedence_1()
+        {
+            UsingNode(SyntaxFactory.ParseExpression("A is B < C, D > [ ]"));
+            N(SyntaxKind.IsExpression);
+            {
+                N(SyntaxKind.IdentifierName);
+                {
+                    N(SyntaxKind.IdentifierToken, "A");
+                }
+                N(SyntaxKind.IsKeyword);
+                N(SyntaxKind.ArrayType);
+                {
+                    N(SyntaxKind.GenericName);
+                    {
+                        N(SyntaxKind.IdentifierToken, "B");
+                        N(SyntaxKind.TypeArgumentList);
+                        {
+                            N(SyntaxKind.LessThanToken);
+                            N(SyntaxKind.IdentifierName);
+                            {
+                                N(SyntaxKind.IdentifierToken, "C");
+                            }
+                            N(SyntaxKind.CommaToken);
+                            N(SyntaxKind.IdentifierName);
+                            {
+                                N(SyntaxKind.IdentifierToken, "D");
+                            }
+                            N(SyntaxKind.GreaterThanToken);
+                        }
+                        N(SyntaxKind.ArrayRankSpecifier);
```

Comment:
Shouldn't this be indented one block less? `ArrayRank` isn't a part of `GenericName`, right?